src/GovUk.Education.ExploreEducationStatistics.Common/Extensions/DbContextExtensions.cs

```diff
@@ -0,0 +1,69 @@
+#nullable enable
+using Microsoft.EntityFrameworkCore;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Extensions
+{
+    public static class DbContextExtensions
+    {
+        /// <summary>
+        /// Reload an entity into the context if it has been detached.
+        /// </summary>
+        /// <remarks>
+        /// If it has been deleted, or cannot be reloaded, we return null as the
+        /// entity would be unsafe to work with.
+        /// <para>
+        /// This is most useful in code where we receive an entity from some other
+        /// service (e.g. in authorization handlers). As we cannot be sure about the
+        /// tracking state of the entity, it is recommended to call this method to
+        /// ensure that it is not stale and has not been deleted.
+        /// </para>
+        /// </remarks>
+        public static T? ReloadEntity<T>(this DbContext context, T entity) where T : class
```

Comment:
Cool, I saw a similar need to introduce something like this after an entity loaded elsewhere with no tracking was then used in an auth handler which failed with an exception when trying to explicitly load related data. I reworked the auth handler and it was no longer an issue but this would have resolved it!

Reply:
Yep :smile: I think it'd be good practice to basically put this in any authorization handler from now on, given we can't necessarily ensure the state of the entity before it arrives there.